base/loading.jl

```diff
@@ -1222,11 +1229,25 @@ function compilecache(pkg::PkgId)
     path === nothing && throw(ArgumentError("$pkg not found during precompilation"))
     return compilecache(pkg, path)
 end
+
+const MAX_NUM_PRECOMPILE_FILES = 10
+
 function compilecache(pkg::PkgId, path::String)
     # decide where to put the resulting cache file
-    cachefile = abspath(DEPOT_PATH[1], cache_file_entry(pkg))
-    cachepath = dirname(cachefile)
+    entrypath, entryfile = cache_file_entry(pkg)
+    cachepath = joinpath(DEPOT_PATH[1], entrypath)
     isdir(cachepath) || mkpath(cachepath)
+    if pkg.uuid === nothing
+       cachefile = abspath(cachepath, entryfile) * ".ji"
+    else
+        candidates = filter!(x -> startswith(x, entryfile * "_"), readdir(cachepath))
+        if length(candidates) < MAX_NUM_PRECOMPILE_FILES
+            idx = length(candidates)
+        else
+            idx = findmin(mtime.(joinpath.(cachepath, candidates)))[2] - 1
```

Comment:
Is this relying on that `candidates` is sorted by `x -> parse(Int, x[length(entryfile * "_") + 1:end - length(".ji")])`?

Reply:
Good point. Newer version should not have this issuel